client/pagination.go

```diff
@@ -0,0 +1,60 @@
+package client
+
+import "strconv"
+
+const limit = 100
+
+type Pagination struct {
+	offset int
+	params map[string]string
+}
+
+type Paginated interface {
+	Environment
```

Comment:
Right now the generics is just for Environment but can be expanded in the future for other types.